tests/integration/test_sns.py

```diff
@@ -894,8 +894,11 @@ def _create_queue(self):
         return queue_name, queue_arn, queue_url
 
     def test_publish_sms_endpoint(self):
+        # Clean posible previous sms messages
+        sns_backend = SNSBackend.get()
```

Comment:
Thanks for this PR @pinzon ! 👍 

Rather than resetting the `sms_messages` list, can we fix the assertion in line 902 - rather than asserting `len` equality, we should assert that all expected messages are contained in the list. (In this case the phone numbers should be reasonably random, and ideally we should not have duplicates across different tests).

We're planning to introduce test parallelization in the near future, and this will help us avoiding undesired side-effects. 👍  Thanks